src/lib/eliom_client.client.ml

```diff
@@ -80,23 +80,55 @@ let int64_to_string i = (Obj.magic i)##toString()
 
 let create_buffer () =
   let elts = ref [] in
-  let add x =
-    elts := x :: !elts
-  in
+  let add x = elts := x :: !elts
+  and get () = List.rev !elts in
   let flush () =
-    let res = List.rev !elts in
+    let res = get () in
     elts := [];
     res
   in
-  add, flush
+  add, get, flush
```

Comment:
I feel we should just use `Queue`, from the stdlib, instead, but it's not very important.